app/scripts/migrations/054.js

```diff
@@ -0,0 +1,58 @@
+import { cloneDeep } from 'lodash';
+
+const version = 54;
+
+/**
+ * Migrates preference tokens with 0 decimals typed as 'string' to 'number'
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+    const state = versionedData.data;
+    const newState = transformState(state);
+    versionedData.data = newState;
+    return versionedData;
+  },
+};
+
+function transformState(state) {
+  const newState = state;
+
+  if (!newState.PreferencesController) {
+    return newState;
+  }
+
+  const { tokens } = newState.PreferencesController || [];
```

Comment:
Nit: The `[]` fallback is unreachable, and isn't the correct type anyway

```suggestion
  const { tokens } = newState.PreferencesController;
```

Reply:
woops! good catch here, I originally had this has `const tokens = newState.preferencesController.tokens || []` but had incorrectly fixed it.

Reply:
fixed